### DISCUSSION TITLE
Help!! Rematch dispatch doesn't  work with me ?

### DISCUSSION THREAD

**Discussion by Oskku** [rematch/rematch#970] [Q&A]
Hei guys !!

I am new with a rematch, I built a rematch model, and I have done all like documentation, but still can't get the data in the reducer array state 

Here are The step I have done 

1. I built model call blogs.js 

`const blogs = {
  state: [], // initial state
  reducers: {
    // handle state changes with pure functions
    blogLoaded:(state, payload) => payload,
    blogDeleted: (state, payload) => {
      state.filter((b) => b.id !== payload.id);
    },
blogAdded: (state, payload) => state.push(payload)
  },
};

export default blogs;

2. I initialize the store with init and add blogs in the models object inside the init function.

`import {init} from '@rematch/core'
import blogs from "./models/blog.model"
import count from "./models/count.model"

const models = {
    blogs,
    count,
  };

const store = init({
    models,
})

export default store;`

3. Then I pass the store in the tree components by the Provider component

`import React from "react";
import ReactDOM from "react-dom/client";
import "./index.css";
import App from "./App";
import { Provider } from "react-redux";
import store from "./store";

const root = ReactDOM.createRoot(document.getElementById("root"));
root.render(
  <Provider store={store}>
    <App />
  </Provider>
);`

4. I built the BlogList component and inside it, I call the connect function  and pass the dispatch and state as props and i call a dummy data Array of objects called blog items and i dispatch it inside the blog loaded function,

import React, { useEffect } from "react";
import { connect } from "react-redux";
import blogItems from "../files/blogItems";

const BlogList = ({ blogLoaded, blogs, blogDeleted }) => {
  console.log(blogItems);
  useEffect(() => {
    blogLoaded(blogItems);
    console.log("UseEffect work");
    // eslint-disable-next-line
  }, []);
  console.log("blogList", blogs);
  return (
    <div className="blog-list">
      <h2>All Blogs items</h2>
      {blogs.map((blog) => (
        <div className="blog-preview" key={blog.id}>
          <h2>{blog.title}</h2>
          <p>Written by {blog.author}</p>
          <button onClick={() => blogDeleted(blog.id)}>Delete</button>
        </div>
      ))}
    </div>
  );
};

const mapState = (state) => ({
  blogs: state.blogs,
});
const mapDispatch = (dispatch) => ({
  blogLoaded: () => dispatch.blogs.blogLoaded,
  blogDeleted:()=> dispatch.blogs.blogDeleted,
});

export default connect(mapState, mapDispatch)(BlogList);

But nothing happened as you see in the picture below? I don't get the dummy data in the state any help ?? did I miss something? 
<img width="673" alt="Screenshot 2022-05-11 at 22 57 07" src="https://user-images.githubusercontent.com/77615959/167935183-09db6af4-6f21-412e-9f73-58ff33a61681.png">

**Comment by semoal** [comment]
hi! You’re not returning the new state in your reducers, which is mandatory if you’re not using @rematch/immer

**Reply by Oskku** [reply]
So how the code should be?  i return it like this but still nothing happend     blogLoaded:(state, payload) => {
      return {
          ...state,
          payload,
      }},

**Reply by Oskku** [reply]
I try with this since yesterday and I really like the rematch, so i hope you could help me with this because for me i don't see any mistake

**Reply by semoal** [reply]
```js
const mapDispatch = (dispatch) => ({
   blogLoaded: () => dispatch.blogs.blogLoaded,
   blogDeleted:()=> dispatch.blogs.blogDeleted,
});
```

These lines are not correct, remove the () => or add a () after the function. 

Next time, please, add a reproduction repository, codesandbox or whatever I can take and reproduce, also the code with some backticks for easier reading. It took me longer to decrypt the pasted code than knew the issue. 

Also, I highly recommend before working with Rematch, understand the basic concepts of Redux and React (You can buy the official guide of Rematch which will be super helpful) 

For example, you don't need mapDispatch or connectProps if you're using functional components you can directly use `useDispatch` and `useSelector`

**Reply by Oskku** [reply]
Thank you so much, I am really good with Redux, also I build redux from scratch https://github.com/Osamah-learn/Build-Redux-From-Scratch so I know the idea of how it works, but Rematch it makes things so easy so I get confused, and yes next time I will put it in sandbox thank you for the advice as you see small things make big problem hahaha,
